cmd/catalyst-node/catalyst-node.go

```diff
@@ -29,35 +30,42 @@ import (
 
 const (
 	httpPort         = 8090
+	httpInternalPort = 8091
 	mistUtilLoadPort = 8042
 )
 
 var Version = "unknown"
+var serfClient *serfclient.RPCClient
```

Comment:
I think this is ok as a global variable. We won't have multiple serf clients per one catalyst node. If you really don't want to have the global variable then you could always have something like `**serfclient.RPCClient` and pass it to `resolveNodeURL()`. But TBH I think it's not worth making it.

One thing that you could actually do is to add a wait with waitgroup before using the `serfClient`. Otherwise you may end up with `panic` if the HTTP endpoint is called before `serfClient` is initialised.